src/test/java/org/jenkinsci/plugins/gitclient/GitAPITestCase.java

```diff
@@ -2120,6 +2120,35 @@ public void test_submodule_checkout_simple() throws Exception {
         assertSubmoduleContents(w.repo);
     }
 
+    /* Opening a git repository in a directory with a symbolic git file instead
+     * of a git directory should function properly. JGit does not implement
+     * renamed submodules, so we the submodules branch won't work with it...
+     */
+    @NotImplementedInJGit
+    public void test_with_repository_works_with_submodule() throws Exception {
+        w = clone(localMirror());
+        assertSubmoduleDirs(w.repo, false, false);
+
+        /* Checkout a branch which includes submodules (in modules directory) */
+        String subBranch = "tests/getSubmodules";
```

Comment:
A JGit compatible submodule test branch is available if you use:

`String subBranch = w.git instanceof CliGitAPIImpl ? "tests/getSubmodules" : "tests/getSubmodules-jgit";`

When I do that, the test fails for JGit without your change, but also fails for JGit with your CliGitAPIImpl change.  I don't yet understand why it fails, since (as you noted) the CliGitAPIImpl and JGitAPIImpl versions of getRepository() are the same once your change is applied.